web/thesauruses/clojure/functions.json

```diff
@@ -0,0 +1,68 @@
+{
+  "meta": {
+    "language": "Clojure",
+    "language_version": "1.10.3",
+    "language_name": "Clojure"
+  },
+  "categories": {
+    "Void Functions": [
+      "void_function_no_parameters",
+      "void_function_with_parameters",
+      "void_function_variable_parameters"
+    ],
+    "Return Value Functions": [
+      "return_value_function_no_parameters",
+      "return_value_function_with_parameters",
+      "return_value_function_variable_parameters"
+    ],
+    "Lambda/Anonymous Functions": [
+      "anonymous_function_no_parameters",
+      "anonymous_function_with_parameters",
+      "anonymous_function_variable_parameters"
+    ],
+    "Subroutines": [
+      "call_subroutine",
+      "return_from_subroutine"
+    ]
+  },
+  "functions": {
+    "void_function_no_parameters": {
+      "code": "(defn void-no-params []\n  (println \"Function called\"))\n\n;; call it\n(void-no-params)\nFunction called",
+      "comment": "There is no concept of void in idiomatic Clojure, so functions with no meaningful return value simply return `nil` (equivalent to `null` in Java)."
+    },
+    "void_function_with_parameters": {
+      "code": "(defn void-params [x y]\n  (println \"x + y is: \" (+ x y)))\n\n;; call it\n(void-with-params 1 2)\nx + y is:  3\nnil"
+    },
+    "void_function_variable_parameters": {
+      "code": "(defn void-variable-params [& params]\n  (println \"Sum of all params: \" (apply + params)))\n\n;; call it\n(void-with-variable-params 1 2 3 4)\nSum of all params:  10\nnil"
```

Comment:
For parts where you say "Sum of all params: 10", I'm fine with this, but can you move it to a `comment` line instead of in the code line? I'm trying to keep `code` lines all technically compileable code.

Reply:
In this particular case, it is compileable/runnable.  The output is a string like `Sum of all params: N` (where N obviously depends on the args).  But I can change it to just return the pure value instead.

Reply:
My point was just to show how you define it here too. So take out hte "call it" and the part on how to do it. This is just how to define the function.